libs/libcommon/src/libcommon/queue.py

```diff
@@ -884,31 +859,30 @@ def finish_job(self, job_id: str, is_success: bool) -> bool:
         except StartedJobError as e:
             logging.error(f"job {job_id} has not the expected format for a started job. Aborting: {e}")
             return False
-        finished_status = Status.SUCCESS if is_success else Status.ERROR
-        previous_status = job.status
-        job.update(finished_at=get_datetime(), status=finished_status)
-        update_metrics_for_type(job_type=job.type, previous_status=previous_status, new_status=finished_status)
+        decrease_metric(job_type=job.type, status=job.status)
+        job.delete()
         release_locks(owner=job_id)
         # ^ bug: the lock owner is not set to the job id anymore when calling start_job()!
         return True
 
-    def cancel_dataset_jobs(self, dataset: str) -> int:
+    def delete_dataset_jobs(self, dataset: str) -> int:
         """
-        Cancel all the jobs for a given dataset.
+        Delete all the jobs for a given dataset.
 
         Args:
             dataset (`str`, required): dataset name
 
         Returns:
             `int`: the number of canceled jobs
         """
+        # TODO: remove status__in=[Status.WAITING, Status.STARTED] filter
         jobs = JobDocument.objects(dataset=dataset, status__in=[Status.WAITING, Status.STARTED])
         previous_status = [(job.type, job.status, job.unicity_id) for job in jobs.all()]
         jobs_to_cancel = len(previous_status)
-        jobs.update(finished_at=get_datetime(), status=Status.CANCELLED)
         for job_type, status, unicity_id in previous_status:
-            update_metrics_for_type(job_type=job_type, previous_status=status, new_status=Status.CANCELLED)
+            decrease_metric(job_type=job_type, status=status)
             release_lock(key=unicity_id)
+        jobs.delete()
```

Comment:
maybe let this row above (instead of the previous `jobs.update`) as I assume we want to release the locks only once the jobs are deleted